string text
string[] word_list
---
bool res